src/blocks/tipset.rs

```diff
@@ -265,6 +266,49 @@ impl Tipset {
             })
         })
     }
+
+    /// Fetch the genesis block header for a given tipset.
+    pub fn genesis(&self, store: impl Blockstore) -> anyhow::Result<BlockHeader> {
+        // Scanning through millions of epochs to find the genesis is quite
+        // slow. Let's use a list of known blocks to short-circuit the search.
+        // The blocks are hash-chained together and known blocks are guaranteed
+        // to have a known genesis.
+        #[derive(Serialize, Deserialize)]
+        struct KnownHeaders {
+            calibnet: HashMap<ChainEpoch, String>,
+            mainnet: HashMap<ChainEpoch, String>,
+        }
+
+        static KNOWN_HEADERS: OnceLock<KnownHeaders> = OnceLock::new();
+        let headers = KNOWN_HEADERS.get_or_init(|| {
+            serde_yaml::from_str(include_str!("../../build/known_blocks.yaml")).unwrap()
```

Comment:
Could we map those strings to cids here? That is to say, have a typed value as soon as possible and avoid `to_string` later.

Reply:
I'll leave that as a future improvement. :)